src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -52,14 +56,33 @@ public void save(
 
         Claim updatedClaim = caseRepository.saveClaimantResponse(claim, response, authorization);
 
-        if (ACCEPTATION == response.getType()) {
-            formaliseResponseAcceptanceService.formalise(updatedClaim, (ResponseAcceptation) response, authorization);
-        }
+        formaliseResponseAcceptance(response, updatedClaim, authorization);
 
         eventProducer.createClaimantResponseEvent(updatedClaim);
         appInsights.trackEvent(getAppInsightsEvent(response), claim.getReferenceNumber());
     }
 
+    private void formaliseResponseAcceptance(ClaimantResponse claimantResponse, Claim claim, String authorization) {
+        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+
+        if (ACCEPTATION == claimantResponse.getType()
+            && !isResponseStatesPaid(response)) {
+            formaliseResponseAcceptanceService.formalise(claim, (ResponseAcceptation) claimantResponse, authorization);
+        }
+    }
+
+    private boolean isResponseStatesPaid(Response response) {
```

Comment:
I think we can move this method in some helper class and write it's unit tests as well. It will be reusable.